### PR TITLE
React Greengrocers

### PR DESCRIPTION
**Pull request by akibria-dev**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,46 +2,75 @@ import './styles/reset.css'
...
   // Setup state here...
+  const [cart, setCart] = useState([])
+  //const [quantity, setQuantity]= useState(1)
 
+  //  Adding item into Cart
+  function addToCart(item) {
+    // check if the item is already in the cart
+    // if it is increase the quantity by 1 and return
+    const itemFound = cart.find(itemFromCart => itemFromCart.id === item.id)
+    if (itemFound) {
```
Comment (georgeroth) [inline src/App.js:22]: Very nice and readable, well done Abul!

File: src/App.js [hunk 2]
```diff
@@ -2,46 +2,75 @@ import './styles/reset.css'
...
+      const others = cart.filter(itemFromCart => itemFromCart.id !== item.id)
+      itemFound.quantity += 1
+      setCart([...others, itemFound])
```
Comment (georgeroth) [inline src/App.js:25]: Will this move the item that you increased the quantity of, to the last place?

File: src/App.js [hunk 3]
```diff
@@ -2,46 +2,75 @@ import './styles/reset.css'
...
+    } else {
+      const cartItem = { ...item, quantity: 1 }
+      setCart([...cart, cartItem])
+    }
+  }
+  // item increase  in the cart
+  const handleIncrement = item => {
+    const updatedCart = cart.map(cartItem => {
```
Comment (georgeroth) [inline src/App.js:33]: Very nice, I think something like this for the above function would be great. Or maybe you can even reuse this code in the above function and just call it!

File: src/App.js [hunk 4]
```diff
@@ -2,46 +2,75 @@ import './styles/reset.css'
...
+  }
+
+  // item decrease in the cart and remove from the list when its 0
+  const handleDecrement = item => {
+    const updatedCart = cart.map(cartItem => {
+      if (item.id === cartItem.id) {
+        const itemFound = { ...cartItem, quantity: --cartItem.quantity }
+        // if the quantity is 0 remove from the list else itemfound, decrease the quantity
+        if (item.quantity === 0) {
+          return null
```
[the remark this answers is not in the document]
Comment (akibria12) [reply src/App.js:49]: well. as you can see after item found then call the function to decrease but when it is less than 1 in quantity its removing from the cart item and displays the rest of the items added in the cart for increasing/decreasing.

File: src/App.js (hunk 2 above)
Comment (akibria12) [reply src/App.js:25]: its filtering the rest of the item in the cart and increasing the item added by 1 each time add to cart function called.

**Pull request closed without merging**